src/v/archival/archival_metadata_stm.h

```diff
@@ -94,6 +94,13 @@ class command_batch_builder {
       cloud_storage::scrub_status status,
       cloud_storage::anomalies detected);
     command_batch_builder& reset_scrubbing_metadata();
+    /// Add read-write fence
+    ///
+    /// The fence prevents all subsequent commands in the batch from being
+    /// applied to the in-memory state of the STM if the 'applied_offset' is
+    /// greater than the 'offset'. This mechanism is supposed to be used as a
```

Comment:
This isn't quite true though, right? It should exactly match the offset?

Also maybe it makes sense to mention this should always be the first record written to an archival batch? Perhaps we could even assert that to be the case.

Reply:
@Lazin can you respond here?